### PR TITLE
Correction for iframe links not working in android

### PR DESCRIPTION
**Pull request by cnico**
## Proposed change

Following the discussion in https://github.com/home-assistant/android/issues/3044, I did some tests after forking the homeassistant frontend code.
As proposed in this Pull Request, if we add the attribute 'allow-top-navigation' on the iframe html sandbox attribute, the problem is solved. With this correction, the behaviour on my android s10 is that when I clic on a link of the iframe's page, it opens in a tab in the default browser of android : this is fine for me instead of having nothing that happens and the user being confused. And it is the same behaviour as in a desktop browser.

Having reproduced the problem both in iframe from the left menu or in iframe from a lovelace card, I corrected it both of them.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Create an iframe card in lovelace with the following src url : https://www.programme-television.org/?no_layout=
Open it on your android device and clic on a link of the tv show : it opens correctly now in a tab of the default android browser.

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/android/issues/3044
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by cnico** [comment]
Hello, 
Any news on this topic ?

File: src/panels/iframe/ha-panel-iframe.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ class HaPanelIframe extends LitElement {
       >
         <iframe
           src=${this.panel.config.url}
-          sandbox="allow-forms allow-popups allow-pointer-lock allow-same-origin allow-scripts allow-modals allow-downloads"
+          sandbox="allow-top-navigation allow-forms allow-popups allow-pointer-lock allow-same-origin allow-scripts allow-modals allow-downloads"
```
Comment (bramkragten) [inline src/panels/iframe/ha-panel-iframe.ts:39]: Can use `allow-top-navigation-by-user-activation`?

**Comment by bramkragten** [comment]
I don't think this is something we should want, an iframe should not take control of the top navigation, that means HA will lose control.... You problem is probably that your TV does not support opening a new tab, and thus will try to set the url on the main tab. I don't think this is a HA issue honestly

**Comment by cnico** [comment]
@bramkragten,
I pushed a new version of my proposal of PR with an explicit choice to the user to activate this behavior of adding allow-top-navigation-by-user-activation to the sandbox attribute.
In this manner, will it be possible to integrate it in HA ?
The security risk of the iframe taking control of the top navigation is in this way not the default and the user knows what he does if he choses to activate it. For me it is better than having no possibility to display the iframe content links and having nothing that happens when the user clics on it.

**Comment by alanmilinovic** [comment]
When will this be merged with the main branch?

**Comment by cnico** [comment]
Here is the new pull request : https://github.com/home-assistant/frontend/pull/15063
@bramkragten : will it be OK for you ?

**Comment by alanmilinovic** [comment]
> Here is the new pull request : #15063 @bramkragten : will it be OK for you ?

Looks good if you ask me.

**Pull request closed without merging**